internal/clirpc/clirpc.go

```diff
@@ -0,0 +1,87 @@
+// clirpc is a package that provides a way to run RPC commands as CLI commands
+package clirpc
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+)
+
+func Invoke(ctx context.Context, handler http.Handler, path, command string, req io.Reader, resp io.Writer) error {
+	grpcBody, err := wrapGRPC(req)
+	if err != nil {
+		return fmt.Errorf("failed to read request: %w", err)
+	}
+	hreq, err := http.NewRequestWithContext(ctx, "POST", "http://localhost"+path+command, grpcBody)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+	hreq.Header.Set("Content-Type", "application/grpc")
+	w := httptest.NewRecorder()
+	handler.ServeHTTP(w, hreq)
+	if w.Code != http.StatusOK {
+		return fmt.Errorf("unexpected status code: %d: %s", w.Code, w.Body.String())
+	}
+	unwrappedResp, err := unwrapGRPC(w.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response: %w", err)
+	}
+	_, err = io.Copy(resp, unwrappedResp)
+	if err != nil {
+		return fmt.Errorf("failed to write response: %w", err)
+	}
+	return nil
+}
+
+// wrapGRPC wraps a proto message for gRPC transport.
+func wrapGRPC(req io.Reader) (io.Reader, error) {
```

Comment:
It's a shame that the `grpc` package doesn't expose proper ways of encoding/decoding the wire format. Doing it manually instead.

Reply:
Connect supports [plain HTTP](https://connectrpc.com/), so you shouldn't really need to do this shenanigans.